### PR TITLE
Clean a little bit Scenario API

### PR DESCRIPTION
**Pull request by cbeauchesne**
## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/_context/_scenarios/auto_injection.py [hunk 1]
```diff
@@ -73,11 +68,6 @@ def __init__(
         if include_centos_7_amd64:
             self.required_vms.append(Centos7amd64())
 
-    def session_start(self):
```
Comment (robertomonteromiguel) [inline utils/_context/_scenarios/auto_injection.py:76]: If we move this code to get_warmups, the method is going to execute several times if we use xdist.

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]

File: utils/_context/_scenarios/auto_injection.py (hunk 1 above)
Comment (robertomonteromiguel) [reply utils/_context/_scenarios/auto_injection.py:76]: Check this: https://github.com/DataDog/system-tests/pull/2970/files

Comment (cbeauchesne) [reply utils/_context/_scenarios/auto_injection.py:76]: Good catch, I've added a `is_main_worker` field. Can you tell me if I've correctly understood what to run here ? 

https://github.com/DataDog/system-tests/pull/2969/files#diff-6bf522167f4c94fdbfb99ce57f86722e8b180e2364bb4315f7f3beb1be479be0R134-R141

File: utils/_context/_scenarios/auto_injection.py [hunk 2]
```diff
@@ -139,21 +128,27 @@ def _check_test_environment(self):
...
-    def _get_warmups(self):
-        logger.terminal.write_sep("=", "Provisioning Virtual Machines", bold=True)
-        return [self.vm_provider.stack_up]
+    def get_warmups(self):
+        warmups = super().get_warmups()
+
+        if self.is_main_worker:
+            warmups.append(lambda: logger.terminal.write_sep("=", "Provisioning Virtual Machines", bold=True))
+
+        warmups.append(self.vm_provider.stack_up)
```
Comment (robertomonteromiguel) [inline utils/_context/_scenarios/auto_injection.py:137]: The stack_up should run only once (in the main worker)

File: utils/_context/_scenarios/auto_injection.py (hunk 1 above)
Comment (robertomonteromiguel) [reply utils/_context/_scenarios/auto_injection.py:76]: Added some comments

File: utils/_context/_scenarios/auto_injection.py [hunk 3]
```diff
@@ -139,21 +128,27 @@ def _check_test_environment(self):
...
                 self._tested_components[key] = vm.tested_components[key].lstrip(" ")
 
-    def pytest_sessionfinish(self, session):
-        logger.info(f"Closing  _VirtualMachineScenario scenario")
-        self.close_targets()
-
     def close_targets(self):
-        logger.info(f"Destroying virtual machines")
+        logger.info("Destroying virtual machines")
         self.vm_provider.stack_destroy()
```
[the remark this answers is not in the document]
Comment (cbeauchesne) [reply utils/_context/_scenarios/auto_injection.py]: done

File: utils/_context/_scenarios/auto_injection.py (hunk 2 above)
Comment (cbeauchesne) [reply utils/_context/_scenarios/auto_injection.py]: done

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**